File: java/Two Pointers/LargestContainerBruteForce.java
public class LargestContainerBruteForce {
    public int largestContainerBruteForce(int[] height) {
        int n = height.length;
        int maxWater = 0;
        // Find the maximum amount of water stored between all pairs of lines.
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                // Calculate the area of the container formed by the lines at indexes i and j.
                int water = Math.min(height[i], height[j]) * (j - i);
                // Update the maximum amount of water stored.
                maxWater = Math.max(maxWater, water);
            }
        }
        return maxWater;
    }
 }

File: kotlin/src/main/kotlin/common/twopointers/largestcontainer/LargestContainerBruteForce.kt
package common.twopointers.largestcontainer

import kotlin.math.max
import kotlin.math.min

class LargestContainerBruteForce {

    fun solution(heights: List<Int>): Int {
        val n = heights.size
        var maxWater = 0
        // Find the maximum amount of water stored between all pairs of lines.
        for (i in 0 until n) {
            for (j in i + 1 until heights.size) {
                val water = min(heights[i], heights[j]) * (j - i)
                maxWater = max(maxWater, water)
            }
        }
        return maxWater
    }
}